### PR TITLE
Adding json-schema compliant validators

### PR DESCRIPTION
**Pull request by allentsai**
No merge

### CODE REVIEW

File: src/PaperG/FirehoundBlob/Facebook/Validators/FacebookAdSetValidator.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    public function validate(FacebookAdSet $adSet) {
+        $data = json_decode(json_encode($adSet->toAssociativeArray()));
+        $this->validator->check($data, (object)['$ref' => 'file://' . realpath(__DIR__ . self::RELATIVE_PATH)]);
+        $result = false;
+        $messages = [];
+        if ($this->validator->isValid()) {
+            $result = true;
+        } else {
+            foreach ($this->validator->getErrors() as $error) {
```
Comment (ktilcu) [inline src/PaperG/FirehoundBlob/Facebook/Validators/FacebookAdSetValidator.php]: `array_map`

**Comment by ktilcu** [comment]
i'm digging the json schema stuff

File: src/PaperG/FirehoundBlob/Facebook/Validators/UnmanagedFacebookBlobValidator.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace PaperG\FirehoundBlob\Facebook\Validators;
+
+use PaperG\FirehoundBlob\Facebook\UnmanagedFacebookBlob;
+use PaperG\FirehoundBlob\JsonValidator;
+
+class UnmanagedFacebookBlobValidator extends JsonValidator
+{
+    const RELATIVE_PATH = '/../../Schema/Facebook/unmanagedFacebookBlobCreate.json';
```
Comment (zhiachong) [inline src/PaperG/FirehoundBlob/Facebook/Validators/UnmanagedFacebookBlobValidator.php]: Do you intend to point relative_path and relative_create_path to the same file?

Comment (allentsai) [inline src/PaperG/FirehoundBlob/Facebook/Validators/UnmanagedFacebookBlobValidator.php]: Hmm, good point, I should probably point at the generic, though it's kind of meaningless...

**Pull request merged**